openedx/core/djangoapps/content_libraries/api/containers.py

```diff
@@ -184,10 +206,20 @@ def _get_container_from_key(container_key: LibraryContainerLocator, isDeleted=Fa
     content_library = ContentLibrary.objects.get_by_key(container_key.lib_key)
     learning_package = content_library.learning_package
     assert learning_package is not None
-    container = authoring_api.get_container_by_key(
-        learning_package.id,
-        key=container_key.container_id,
-    )
+
+    container: Container
+    match container_key.container_type:
+        case ContainerType.Unit.value:
+            container = authoring_api.get_unit_by_key(
+                learning_package.id,
+                key=container_key.container_id,
+            )
+        case _:
+            container = authoring_api.get_container_by_key(
+                learning_package.id,
+                key=container_key.container_id,
+            )
```

Comment:
Not sure if we need the `get_unit_by_key`. You can access the unit using `container.unit`.